pkg/parser/types/contract/contracts.go

```diff
@@ -21,15 +22,44 @@ type Contract interface {
 	GetType() string
 }
 
-type Resource struct {
-	gotypes.DataType
-	Package string
-	Name    string
+// Common parts for all contracts
+type CommonData struct {
+	// The name of package where contract was made.
+	Package            string
+	// The expected type for this contract.
+	ExpectedType       gotypes.DataType
+	// Is true if the expected data type for this contract was derived
+	// according to the Golang rules.
+	DataTypeWasDerived bool
+}
+
+// Contract for function definitions/declarations.
+//
+// From a Golang point of view
+//
+//   f := func() {}	and	func f() {}
+//
+// are different constructs. Both introduces a new symbol f, but the ways are
+// different and we want to keep these ways.
+type Function struct {
```

Comment:
Let's use ``FunctionLiteral`` so we have all three literals explicitly in the same category